lib/common/foundation/api_response_v2.dart

```diff
@@ -0,0 +1,151 @@
+/// Standardized API response wrapper for all API calls
+/// 
+/// This class provides a unified way to handle success and error states
+/// from API responses, replacing the inconsistent ApiResult and ApiResponse classes.
+class ApiResponseV2<T> {
+  /// The response data, null if error occurred
+  final T? data;
+  
+  /// Whether the API call was successful
+  final bool isSuccess;
+  
+  /// Error message if the call failed
+  final String? errorMessage;
+  
+  /// HTTP status code from the response
+  final int? statusCode;
```

Comment:
As our server end won't provide statusCode as part of the response, we can safely remove it from ApiResponseV2